evm/send-tokens.js

```diff
@@ -31,17 +62,80 @@ async function sendTokens(chain, options) {
         if (anwser !== 'y') return;
     }
 
+    let gasLimit, gasPrice;
+
+    try {
+        gasPrice = parseUnits((await provider.getGasPrice()).toString(), 'gwei');
+        const block = await provider.getBlock('latest');
+        gasLimit = block.gasLimit.toNumber() / 1000;
+
+        printInfo('Gas Price:', gasPrice.toString());
+        printInfo('Gas Limit:', gasLimit.toString());
+    } catch (error) {
+        printError('Gas price and limit could not be fetched from provider');
+        printObj(error);
+    }
+
+    let nonce = await getNonceFromProvider(provider, signerAddress);
+    let signersData, transactions;
+
+    if (isOffline) {
+        printInfo(`Storing signed Txs offline in file ${fileName}`);
+        directoryPath = directoryPath || './txs';
+        fileName = fileName || env.toLowerCase() + '-' + chain.name.toLowerCase() + '-' + 'signedSendTokensTransactions';
+        nonce = await getLatestNonceAndUpdateData(directoryPath, fileName, wallet);
+        signersData = await getAllSignersData(directoryPath, fileName);
+        transactions = await getTransactions(directoryPath, fileName, signerAddress);
+    }
+
     for (const recipient of recipients) {
         printInfo('Recipient', recipient);
 
-        const tx = await wallet.sendTransaction({
-            to: recipient,
-            value: amount,
-        });
+        if (privateKey === 'ledger') {
+            const { baseTx, signedTx } = await ledgerSign(gasLimit, gasPrice, nonce, chain, wallet, recipient, amount);
+
+            if (isOffline) {
+                const tx = {};
+                tx.nonce = nonce;
+                tx.msg = `This transaction will send ${amount} of native tokens to ${recipient} on chain ${chain.name} with chainId ${chain.chainId}`;
+                tx.baseTx = baseTx;
+                tx.signedTx = signedTx;
+                tx.status = 'PENDING';
+                transactions.push(tx);
+            } else {
+                try {
+                    const response = await sendTx(signedTx, provider);
+
+                    if (response.error || !isValidJSON(response) || response.status !== 1) {
+                        const error = `Execution failed${
+                            response.status ? ` with txHash: ${response.transactionHash}` : ` with msg: ${response.message}`
+                        }`;
+                        throw new Error(error);
+                    }
+
+                    printInfo('Transaction hash', response.transactionHash);
+                } catch (error) {
+                    printError('Broadcasting Transaction failed');
+                    printObj(error);
+                }
+            }
```

Comment:
hide details inside `sendTx`? It can print any errors, and return the tx hash, receipt, status if caller needs to know